src/services/service.device.js

```diff
@@ -0,0 +1,107 @@
+const axios = require("axios");
+const config = require("../config");
+
+const baseURL = config.base_local_url_graphql;
+const getHeader = (token) => ({
+  headers: {'content-type': 'application/json', Authorization: `${token}`},
+})
+
+const getDeviceById = (token, id) => {
+  return axios.get(`${baseURL}/device/${id}`, getHeader(token))
+}
+
+const getDeviceList = async (token, ids) => {
+  const promises = [];
+  const values = {};
+  ids.forEach(deviceId => {
+    const promise = axios.get(`${baseURL}/device/${deviceId}`, getHeader(token)).then((response) => {
+      if (!!response.data) {
+        const {data: {attrs, created, id, label, templates}} = response;
+        values[id] = {attrs, created, id, label, templates, templateID: null}
+      }
+    }).catch(() => Promise.resolve(null));
+    promises.push(promise);
+  })
+
+  await (Promise.all(promises));
```

Comment:
It sounds a little strange to use the "values" variable as a closure for the callback, but It's okay.
I think these could be done with only one array (returning to values[id] and after Promises.all(values[id]) );